src/compiler/utilities.ts

```diff
@@ -2016,6 +2016,8 @@ namespace ts {
             case SyntaxKind.AwaitExpression:
             case SyntaxKind.MetaProperty:
                 return true;
+            case SyntaxKind.ExpressionWithTypeArguments:
+                return !(isPartOfTypeNode(node.parent) || isHeritageClause(node.parent));

```

Comment:
Why is it not an expression in a heritage clause? Feels like it must be the case that it's an expression in an `extends` clause, right? @ahejlsberg

Reply:
The `isPartOfTypeNode(node.parent)` implies that we're in a heritage clause (of an interface), so it's redundant here. In other words, the code is equivalent to just `return !isHeritageClause(node.parent)`. That said, it's probably more correct to change the code to `return !isPartOfTypeNode(node.parent)`, which would say that if it's not a type, it must be an expression. That would additionally return true when the expression with type arguments occurs in a class heritage clause.

Reply:
So just

```suggestion
                return !isPartOfTypeNode(node.parent);
```

?

Reply:
@andrewbranch commented that that's likely desirable for classes, but not necessarily for types.

Reply:
I used [this example](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEA3W8BCUDOIBc8De8AdiAO4A8AKgHwAUAlLlIQJ7wC+AUAC7MAOC6LPAC88AEQALKPCnwMEgPYBXCMHhcJASwzwARgg0G+CKF3gADEAA8uIQsB2CQ5gPxiA3Bw6RMOgMLw1rb2jpggZBhcMJqEAOZU+F6cQA) to demonstrate why using `getTypeFromTypeNode` would be wrong there. If something is an expression, you have to get the type of the expression.